workshop/Lab3/README.md

```diff
@@ -0,0 +1,252 @@
+	
+- Create the pod with label app=myguestbook using pod.yaml 
+
+   ```kubectl create -f guestbook-deployment.yaml```
+
+- List the pod with label app=guestbook 
+
+   ```kubectl get pods -l app=guestbook```
+
+2. We now define a service object to expose the deployment to external clients.
+```
+	apiVersion: v1
+	kind: Service
+	metadata:
+	  name: guestbook
+	  labels:
+	    app: guestbook
+	spec:
+	  ports:
+	  - port: 3000
+	    targetPort: http-server
+	  selector:
+	    app: guestbook
+	  type: LoadBalancer
+``` 
+The above configuration allows to create a service object named 'guestbook'  which will allow incoming trafiic for guestbook to port 3000 on one of the running pods. Port type 'LoadBalancer' is one of the types that exposes service using cloud providers loadbalancer to route trafiic to Nodeport or ClusterIP.
+
+- Let us now create guestbook service.
+		```kubectl create -f guestbook-service.yaml```
+		
+- Test guestbook app using a browser of your choice using the url '\<your-cluster-ip\>:\<node-port\>'
+
+You can simply change the deployment, alter replicas using the following command. Kubernetes then sees the desired the you want and tries to reconcile the actual current state to the desired stat. 
+
+```	
+	kubectl edit deployment guestbook 
+```
+
+# 4. Connect to a back-end service.
+
+As you have noticed that guestbook app default to in-memory database when it cannot see a redis service running in the cluster. By doing so we can realize that  our data is not consistent. The data returned will change depending on which pod  running our guestbook application, in the cluster, is serving our request. 
+In order to achieve data consistency we should have a central database server that manages the data. 
+Let us now connect to a back-end database and have guestbook persist from data. We can do this declaratively by using a configuration file to deploy the redis app.
+
+Our first application is a simple 3-tier application
+
+1. Deploy redis database named 'redis-master', this will create a single instance of a database, replicas set to 1, that will server the guestbook app instances to persist data and read data from. This application runs the image 'redis:2.8.23' and exposes redis port 6379.
+
+```
+	apiVersion: apps/v1
+	kind: Deployment
+	metadata:
+	  name: redis-master
+	  labels:
+	    app: redis
+	    role: master
+	spec:
+	  replicas: 1
+	  selector:
+	    matchLabels:
+	      app: redis
+	      role: master
+	  template:
+	    metadata:
+	      labels:
+	        app: redis
+	        role: master
+	    spec:
+	      containers:
+	      - name: redis-master
+	        image: redis:2.8.23
+	        ports:
+	        - name: redis-server
+	          containerPort: 6379
+```
+
+- Create the pod  running redis deployment 
+	``` kubectl create -f redis-master-deployment.yaml ```
+	
+- check to see that redis server pod is running
+	``` 
+		kubectl get pods -lapp=redis,role=master
```

Comment:
Do me a favor and add `$ ` to each command so that the command lines are very clearly differentiated from the output. please add it to all commands, even if you don't show the output so we're consistent.